### PR TITLE
batch_norm_jvp: improve error message when running_{mean,var} have forward grad defined

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73655
* #72811
* #72677

Fixes: https://github.com/pytorch/pytorch/issues/73541

Differential Revision: [D34586758](https://our.internmc.facebook.com/intern/diff/D34586758)

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Thanks!

File: torch/csrc/autograd/FunctionsManual.cpp [hunk 1]
```diff
@@ -4841,6 +4841,9 @@ Tensor batch_norm_jvp(
     TORCH_INTERNAL_ASSERT(
         running_mean.has_value() && running_var.has_value(),
         "Expect running_mean and running_var to have value when train=false");
+    TORCH_CHECK(
+        !running_mean.value()._fw_grad(/*level=*/0).defined() && !running_mean.value()._fw_grad(/*level=*/0).defined(),
+        "batch_norm_jvp expects running_mean and running_var to not have forward grad defined");
```
Comment (zou3519) [inline torch/csrc/autograd/FunctionsManual.cpp:4846]: Nit: "running_mean and running_var are not differentiable, you attempted to differentiate them while performing forward-mode AD on batch_norm" would be clearer to the user?

**Review by zou3519 — APPROVED** [review]
LGTM, minor nit and also we should probably add a test for this to prevent regressions (e.g. someone refactoring batch_norm_jvp and accidentally deleting those two lines)

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34586758).

**Pull request closed without merging**